src/Initializer/BatchRecorders/LocalIntegrationRecorder.cpp

```diff
@@ -171,8 +171,160 @@ void LocalIntegrationRecorder::recordDisplacements() {
+
+        (*currentTable)[key].set(inner_keys::Wp::Id::Derivatives, derivatives[face]);
+        (*currentTable)[key].set(inner_keys::Wp::Id::Dofs, dofsPtrs[face]);
+        (*currentTable)[key].set(inner_keys::Wp::Id::Idofs, idofsPtrs[face]);
+        (*currentTable)[key].set(inner_keys::Wp::Id::AminusT, aminusTPtrs[face]);
+
+        (*currentTable)[key].set(inner_keys::Wp::Id::T, T[face]);
+        (*currentTable)[key].set(inner_keys::Wp::Id::Tinv, Tinv[face]);
+
+        (*currentTable)[key].set(inner_keys::Wp::Id::FaceDisplacement, displacementsPtrs[face]);
+        (*currentMaterialTable)[key].set(inner_keys::Material::Id::Rho, rhos[face]);
+        (*currentMaterialTable)[key].set(inner_keys::Material::Id::Lambda, lambdas[face]);
+
+        (*currentTable)[key].set(inner_keys::Wp::Id::NodalAvgDisplacements,
+                                 nodalAvgDisplacementsPtrs[face]);
+      }
+    }
+  }
+}
+
+void LocalIntegrationRecorder::recordDirichletBc() {
+  const auto size = currentLayer->getNumberOfCells();
+  if (size > 0) {
+    std::array<std::vector<real*>, 4> dofsPtrs{};
+    std::array<std::vector<real*>, 4> idofsPtrs{};
+    std::array<std::vector<real*>, 4> Tinv{};
+    std::array<std::vector<real*>, 4> aminusTPtrs{};
+
+    std::array<std::vector<real*>, 4> easiBoundaryMapPtrs{};
+    std::array<std::vector<real*>, 4> easiBoundaryConstantPtrs{};
+
+    for (unsigned cell = 0; cell < size; ++cell) {
+      auto data = currentLoader->entry(cell);
+
+      for (unsigned face = 0; face < 4; ++face) {
+        if (data.cellInformation.faceTypes[face] == FaceType::dirichlet) {
+
+          dofsPtrs[face].push_back(static_cast<real*>(data.dofs));
+          idofsPtrs[face].push_back(idofsAddressRegistry[cell]);
+
+          Tinv[face].push_back(data.boundaryMapping[face].TinvData);
+          aminusTPtrs[face].push_back(data.neighIntegrationOnDevice.nAmNm1[face]);
+
+          easiBoundaryMapPtrs[face].push_back(data.boundaryMapping[face].easiBoundaryMap);
+          easiBoundaryConstantPtrs[face].push_back(data.boundaryMapping[face].easiBoundaryConstant);
+        }
+      }
+    }
+
+    for (unsigned face = 0; face < 4; ++face) {
+      if (!dofsPtrs[face].empty()) {
+        ConditionalKey key(*KernelNames::BoundaryConditions, *ComputationKind::Dirichlet, face);
+        checkKey(key);
+        (*currentTable)[key].set(inner_keys::Wp::Id::Dofs, dofsPtrs[face]);
+        (*currentTable)[key].set(inner_keys::Wp::Id::Idofs, idofsPtrs[face]);
+        (*currentTable)[key].set(inner_keys::Wp::Id::AminusT, aminusTPtrs[face]);
+        (*currentTable)[key].set(inner_keys::Wp::Id::Tinv, Tinv[face]);
+
+        (*currentTable)[key].set(inner_keys::Wp::Id::EasiBoundaryMap, easiBoundaryMapPtrs[face]);
+        (*currentTable)[key].set(inner_keys::Wp::Id::EasiBoundaryConstant,
+                                 easiBoundaryConstantPtrs[face]);
+      }
+    }
+  }
+}
+
+void LocalIntegrationRecorder::recordAnaliticalBc() {
```

Comment:
typo in function name, should be recordAnalyticalBc

Reply:
done